### DISCUSSION TITLE
Why not namespace the components like they do in the React version?

### DISCUSSION THREAD

**Discussion by Evertt** [rgossiaux/svelte-headlessui#66] [Q&A] [answered]
In the readme you write:

> Components do not have . in their names; use ListboxButton instead of Listbox.Button

And I'm wondering, why though? It's 100% possible to do it the same way as the React version. See here:
https://svelte.dev/repl/0899a509e81f472790133a21a3ac949f?version=3.46.4

And I think it would be nicer to namespace the components like that, because then your import statements can be a lot shorter.

```js
import { Menu, MenuButton, MenuItems, MenuItem, Listbox, ListboxButton, ListboxOptions } from '@rgossiaux/svelte-headlessui'
```

Could then just be

```js
import { Menu, Listbox } from '@rgossiaux/svelte-headlessui'
```

**Comment by rgossiaux** [comment, marked as answer]
The React version uses dots, the Vue version does not. I went with the Vue naming as it felt closer more "Svelte-y" to me. I don't think optimizing for the number of characters in your import statement is an important consideration.

**Reply by Evertt** [reply]
> I don't think optimizing for the number of characters in your import statement is an important consideration.

Why not? I understand that it shouldn't be the highest priority. But fewer characters can help with reading and interpreting the code later.

**Reply by slinso** [reply]
Another point is copying code from headless UI. I always use the React code, as it is closer to your implementation and it is easier to adapt to svelte-headlessui. So by using the dot notation it gets even closer.

**Reply by vhscom** [reply]
I'm very pleased with the design decision. Adding dots inside component names isn't not just Svelte-y it's not Web-y.

**Comment by greendesertsnow** [comment]
Transition enterFrom, enterTo and leaveFrom do not feel Svelty either, they're not weby. They should have had kabob-case. 
This seems inconsistent with the some parts of the documentation. enter-from is more weby and easier to copy from the Vue code.

It says in the documentation: SVG attributes in React like strokeWidth, strokeLinecap, etc. become stroke-width, stroke-linecap, etc. in Svelte.

https://svelte-headlessui.goss.io/docs/tailwind-ui#jsx-camelcased-attribute-names

**Reply by rgossiaux** [reply]
There's already an open issue to rename them: https://github.com/rgossiaux/svelte-headlessui/issues/57
